### PR TITLE
Bay12 examine overhaul

### PR DESCRIPTION
**Pull request by tigercat2000**
This commit overhauls the examine system to baystation's latest system,
including a more efficient verb approach, and a new status panel tab,
which shows more information for pre-defined objects.

basic
drone
Antagonists can have special text for certain objects.
drone_a

Huge amount of file changes are simply converting to the new examine standard.
Something else fun, a few examine() procs were broken with our old system. Now they aren't. Notable things include, Seeing a blob's chemical type, Operative pinpointers showing you where the closest operative is, Radios listing all of their active encryption key channels.

### CODE REVIEW

**Comment by Alex6511** [comment]
:+1:

**Comment by Spacemanspark** [comment]
:+1:

File: code/modules/examine/descriptions/weapons.dm [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	Note: This file is meant for actual weapons (guns, swords, etc), and not the stupid 'every obj is a weapon, except when it's not' thing.
+*/
+
+//******
+//*Guns*
+//******
+
+//This contains a lot of copypasta but I'm told it's better then a lot of New()s appending the var.
+/obj/item/weapon/gun
+	description_info = "This is a gun.  To fire the weapon, ensure your intent is *not* set to 'help', have your gun mode set to 'fire', \
```
Comment (Regen1) [inline code/modules/examine/descriptions/weapons.dm]: I'm not sure if this is a thing on paradise, I think guns still fire even when on help intent

Comment (tigercat2000) [inline code/modules/examine/descriptions/weapons.dm]: Yeah, they do. Copypasta for the win!

**Comment by Fox-McCloud** [comment]
Those custom descriptions for some of the things are obnoxiously long.

In any event, I'm guessing this is the mob verb based examine? If that's the case, could you please put it back the right click menu?

**Comment by sasanek12** [comment]
:+1:

**Comment by Spacemanspark** [comment]
Why put things in the right click menu, it's much faster to just shift + click on what you want to examine.

**Comment by tigercat2000** [comment]
@Fox-McCloud Actually, it should already be in the right click menu, because I didn't actually remove it from it.

And, because the examine tab is literally it's own full tab which is something like 440x600 pixels, it can afford huge text.

**Comment by Fox-McCloud** [comment]
Ok, great. 

@Spacemanspark 

Because it allows you to see what's in a stack of items at range---it also lets you see if something is hiding under a table.

The only reason we got rid of right click examine is because of performance issues; you would right click on a stack of items and freeze forever. 

This changes examining from an atom verb to a mob verb which calls a proc on atoms--this makes it a LOT cheaper and incredibly quick, so it won't lag players.

File: code/modules/examine/descriptions/medical.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	Right-click the scanner and click 'Eject Occupant' to remove them.  You can enter the scanner yourself in a similar way, using the 'Enter Body Scanner' \
+	verb."
+
+/obj/machinery/atmospherics/unary/cryo_cell
+	description_info = "The cryogenic chamber, or 'cryo', treats most damage types, most notably genetic damage. It also stabilizes patients \
+	in critical condition by placing them in stasis, so they can be treated at a later time.<br>\
+	<br>\
+	In order for it to work, it must be loaded with chemicals, and the temperature of the solution must reach a certain point. Additionally, it \
+	requires a supply of pure oxygen, provided by canisters that are attached. The most commonly used chemicals in the chambers are Cryoxadone and \
+	Clonexadone. Clonexadone is more effective in treating all damage, including Genetic damage, but is otherwise functionally identical.<br>\
```
Comment (Fox-McCloud) [inline code/modules/examine/descriptions/medical.dm]: clonex doesn't exist anymore; there's only Cryox

File: code/modules/examine/descriptions/weapons.dm [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+//This contains a lot of copypasta but I'm told it's better then a lot of New()s appending the var.
+/obj/item/weapon/gun
+	description_info = "This is a gun.  To fire the weapon, have your gun mode set to 'fire', \
+	then click where you want to fire."
+
+/obj/item/weapon/gun/energy
+	description_info = "This is an energy weapon.  To fire the weapon, have your gun mode set to 'fire', \
+	then click where you want to fire.  Most energy weapons can fire through windows harmlessly.  To recharge this weapon, use a weapon recharger."
+
+/obj/item/weapon/gun/energy/crossbow
```
Comment (Fox-McCloud) [inline code/modules/examine/descriptions/weapons.dm]: this needs to be tagged to the kinetic accelerator variety, not this one; this is currently an admin only weapon.

File: code/modules/examine/descriptions/weapons.dm [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	then click where you want to fire."
+	description_antag = "This is a stealthy weapon which fires poisoned bolts at your target.  When it hits someone, they will suffer a stun effect, in \
+	addition to toxins.  The energy crossbow recharges itself slowly, and can be concealed in your pocket or bag."
+
+/obj/item/weapon/gun/energy/gun
+	description_info = "This is an energy weapon.  To fire the weapon, have your gun mode set to 'fire', \
+	then click where you want to fire.  Most energy weapons can fire through windows harmlessly.  To switch between stun and lethal, click the weapon \
+	in your hand.  To recharge this weapon, use a weapon recharger."
+
+/obj/item/weapon/gun/energy/gun/taser
```
Comment (Fox-McCloud) [inline code/modules/examine/descriptions/weapons.dm]: Not accessible to players; might want to make this the hybrid taser and ensure that the description mentions it has a stun and a disable mode.

File: code/modules/examine/descriptions/weapons.dm [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	description_info = "This is an energy weapon.  To fire the weapon, have your gun mode set to 'fire', \
+	then click where you want to fire.  Most energy weapons can fire through windows harmlessly. To recharge this weapon, use a weapon recharger."
+
+/obj/item/weapon/gun/energy/gun/stunrevolver
```
Comment (Fox-McCloud) [inline code/modules/examine/descriptions/weapons.dm]: One thing I'm noticing; these mention being able to fire through windows, which, on Bay, is correct...but isn't here for anything other than beam weapons.

Comment (tigercat2000) [inline code/modules/examine/descriptions/weapons.dm]: I assumed the "fire" mode meant "make sure you don't have targeting mode on if you want to fire"

**Comment by Fox-McCloud** [comment]
@tigercat2000 

Merge conflicts. =(

**Comment by tigercat2000** [comment]
I wish github actually told you what was conflicting without having to try updating and see what happens

**Pull request merged**